pkg/transportrequest/uploadcts.go

```diff
@@ -0,0 +1,233 @@
+	}
+
+	deployStatement, err := getFioriDeployStatement(action.TransportRequestID, action.ConfigFile, action.Application, action.Connection)
+	if err != nil {
+		return err
+	}
+
+	cmd = append(cmd, deployStatement)
+
+	return command.RunShell("/bin/bash", strings.Join(cmd, "\n"))
+}
+
+func getPrepareFioriEnvironmentStatement(deps []string, npmInstallOpts []string) string {
+	cmd := []string{
+		"npm",
+		"install",
+		"--global",
+	}
+	cmd = append(cmd, npmInstallOpts...)
+	cmd = append(cmd, deps...)
+	return strings.Join(cmd, " ")
+}
+
+func getFioriDeployStatement(
+	transportRequestID string,
+	configFile string,
+	app CTSApplication,
+	cts CTSConnection,
+) (string, error) {
+	desc := app.Desc
+	if len(desc) == 0 {
+		desc = "Deployed with Piper based on SAP Fiori tools"
+	}
+
+	useConfigFile, noConfig, err := handleConfigFile(configFile)
+	if err != nil {
+		return "", err
+	}
+	cmd := []string{
+		"fiori",
+		"deploy",
+		"--failfast", // provide return code != 0 in case of any failure
+		"--yes",      // autoconfirm --> no need to press 'y' key in order to confirm the params and trigger the deployment
+		"--username", abapUserKey,
+		"--password", abapPasswordKey,
+		"--description", fmt.Sprintf("\"%s\"", desc),
+	}
+
+	if noConfig {
+		cmd = append(cmd, "--noConfig") // no config file, but we will provide our parameters
+	}
+	if useConfigFile {
+		cmd = append(cmd, "--config", fmt.Sprintf("\"%s\"", configFile))
+	}
+	if len(cts.Endpoint) > 0 {
+		log.Entry().Debugf("Endpoint '%s' used from piper config", cts.Endpoint)
+		cmd = append(cmd, "--url", cts.Endpoint)
+	} else {
+		log.Entry().Debug("No endpoint found in piper config.")
+	}
+	if len(cts.Client) > 0 {
+		log.Entry().Debugf("Client '%s' used from piper config", cts.Client)
+		cmd = append(cmd, "--client", cts.Client)
+	} else {
+		log.Entry().Debug("No client found in piper config.")
+	}
+	if len(transportRequestID) > 0 {
+		log.Entry().Debugf("TransportRequestID '%s' used from piper config", transportRequestID)
+		cmd = append(cmd, "--transport", transportRequestID)
+	} else {
+		log.Entry().Debug("No transportRequestID found in piper config.")
+	}
+	if len(app.Pack) > 0 {
+		log.Entry().Debugf("application package '%s' used from piper config", app.Pack)
+		cmd = append(cmd, "--package", app.Pack)
+	} else {
+		log.Entry().Debug("No application package found in piper config.")
+	}
+	if len(app.Name) > 0 {
+		log.Entry().Debugf("application name '%s' used from piper config", app.Name)
+		cmd = append(cmd, "--name", app.Name)
+	} else {
+		log.Entry().Debug("No application name found in piper config.")
+	}
+
+	return strings.Join(cmd, " "), nil
+}
+
+func getSwitchUserStatement(user string) string {
+	return fmt.Sprintf("su %s", user)
+}
+
+func handleConfigFile(path string) (useConfigFile, noConfig bool, err error) {
+
+	if len(path) == 0 {
+		exists, e := files.FileExists(defaultConfigFileName)
```

Comment:
> no need for a new error `e`

Without the `e` I get an error message from the compile: `err is shadowed during return`.

@o-liver : Does this work on your end?

Reply:
Oh, I have to check

Reply:
Yes, I get the same. It is because inside the `if len(path) == 0` closure there is a new scope and with `:=` a new err var is created. By defining exists before and changing `:=` to `=` it works... Not very elegant either, so you do what you like more.